--- a/plugins/simplestream/simplestream.cc
+++ b/plugins/simplestream/simplestream.cc
@@ -104,6 +104,7 @@ class Simple_Stream : public Plugin_Api {
             json json_object;
             std::string json_string;
             std::vector<boost::asio::const_buffer> send_buffer;
+            uint32_t json_length = 0;
             if (stream.sendJSON==true){
               //create JSON metadata
               json_object = {
@@ -117,7 +118,7 @@ class Simple_Stream : public Plugin_Api {
                  {"event","audio"},
               };
               json_string = json_object.dump();
-              uint32_t json_length = json_string.length();  //determine length in bytes
+              json_length = json_string.length();  //determine length in bytes
               //BOOST_LOG_TRIVIAL(debug) << "json_length is " <<json_length <<" bytes";
               send_buffer.push_back(buffer(&json_length,4));  //prepend length of the json data
               send_buffer.push_back(buffer(json_string));  //prepend json data
@@ -180,6 +181,7 @@ class Simple_Stream : public Plugin_Api {
               json json_object;
               std::string json_string;
               std::vector<boost::asio::const_buffer> send_buffer;
+              uint32_t json_length = 0;
               if (stream.sendJSON==true){
                 //create JSON metadata
                 json_object = {
@@ -194,7 +196,7 @@ class Simple_Stream : public Plugin_Api {
                    {"event","call_start"},
                 };
                 json_string = json_object.dump();
-                uint32_t json_length = json_string.length();  //determine length in bytes
+                json_length = json_string.length();  //determine length in bytes
                 send_buffer.push_back(buffer(&json_length,4));  //prepend length of the json data
                 send_buffer.push_back(buffer(json_string));  //prepend json data
               }
@@ -231,6 +233,7 @@ class Simple_Stream : public Plugin_Api {
               json json_object;
               std::string json_string;
               std::vector<boost::asio::const_buffer> send_buffer;
+              uint32_t json_length = 0;
               if (stream.sendJSON==true){
                 //create JSON metadata
                 json_object = {
@@ -241,7 +244,7 @@ class Simple_Stream : public Plugin_Api {
                    {"event","call_end"},
                 };
                 json_string = json_object.dump();
-                uint32_t json_length = json_string.length();  //determine length in bytes
+                json_length = json_string.length();  //determine length in bytes
                 send_buffer.push_back(buffer(&json_length,4));  //prepend length of the json data
                 send_buffer.push_back(buffer(json_string));  //prepend json data
               }
